src/prefect/engine/cloud/task_runner.py

```diff
@@ -169,7 +169,8 @@ def check_task_is_cached(self, state: State, inputs: Dict[str, Result]) -> State
         if self.task.cache_for is not None:
             oldest_valid_cache = datetime.datetime.utcnow() - self.task.cache_for
             cached_states = self.client.get_latest_cached_states(
-                task_id=self.task.id, created_after=oldest_valid_cache
+                task_id=prefect.context.get("task_id", ""),
```

Comment:
Is this still valid?  It looks like you aren't populating `task_id` in context anymore?

Reply:
It's placed into context when the flow run is initialized: https://github.com/PrefectHQ/prefect/pull/940/files#diff-440efc6cfb70f74bea6228c12fb7dc2fR170

Reply:
You removed that though, because `task.id` is no longer an attribute

Reply:
Yes, it's not taken from `task.id` anymore but instead when we query Cloud for all task run states, we also get their IDs to make future queries easier. So it's being set from the result of that query.